x-pack/plugins/observability_solution/synthetics/public/apps/synthetics/components/monitor_details/monitor_status/use_monitor_status_data.ts

```diff
@@ -5,75 +5,132 @@
  * 2.0.
  */
 
-import { useCallback, useMemo, useState } from 'react';
-import { throttle } from 'lodash';
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { useDebounce } from 'react-use';
+import { useLocation } from 'react-router-dom';
 
-import { scheduleToMinutes } from '../../../../../../common/lib/schedule_to_time';
 import { useSyntheticsRefreshContext } from '../../../contexts/synthetics_refresh_context';
 
 import { useSelectedMonitor } from '../hooks/use_selected_monitor';
-import { usePingStatuses } from '../hooks/use_ping_statuses';
 import {
   dateToMilli,
   createTimeBuckets,
-  createStatusTimeBins,
   CHART_CELL_WIDTH,
   indexBinsByEndTime,
   MonitorStatusPanelProps,
+  createStatusTimeBins,
+  MonitorStatusTimeBin,
 } from './monitor_status_data';
+import { useSelectedLocation } from '../hooks/use_selected_location';
+import {
+  clearMonitorStatusHeatmapAction,
+  quietGetMonitorStatusHeatmapAction,
+  selectHeatmap,
+} from '../../../state/status_heatmap';
+
+type Props = Pick<MonitorStatusPanelProps, 'from' | 'to'> & { initSize?: number };
 
-export const useMonitorStatusData = ({
-  from,
-  to,
-}: Pick<MonitorStatusPanelProps, 'from' | 'to'>) => {
+export const useMonitorStatusData = ({ from, to, initSize }: Props) => {
   const { lastRefresh } = useSyntheticsRefreshContext();
   const { monitor } = useSelectedMonitor();
-  const monitorInterval = Math.max(3, monitor?.schedule ? scheduleToMinutes(monitor?.schedule) : 3);
+  const location = useSelectedLocation();
+  const pageLocation = useLocation();
 
   const fromMillis = dateToMilli(from);
   const toMillis = dateToMilli(to);
   const totalMinutes = Math.ceil(toMillis - fromMillis) / (1000 * 60);
-  const pingStatuses = usePingStatuses({
-    from: fromMillis,
-    to: toMillis,
-    size: Math.min(10000, Math.ceil((totalMinutes / monitorInterval) * 2)), // Acts as max size between from - to
-    monitorInterval,
+
+  const [binsAvailableByWidth, setBinsAvailableByWidth] = useState<number | null>(null);
+  const [debouncedBinsCount, setDebouncedCount] = useState<number | null>(null);
+
+  const minsPerBin =
+    debouncedBinsCount !== null ? Math.floor(totalMinutes / debouncedBinsCount) : null;
+
+  const dispatch = useDispatch();
+  const { heatmap: dateHistogram, loading } = useSelector(selectHeatmap);
+
+  useEffect(() => {
+    if (binsAvailableByWidth === null && initSize) {
```

Comment:
Since `initSize` comes from a `ref`, is there a chance that this value will be outdated by the time this `useEffect` runs? I'm curious because if the element resizes as child components finish loading, the `initValue` could be wrong